cpp/oneapi/dal/backend/memory.hpp

```diff
@@ -80,6 +82,161 @@ class usm_deleter {
     sycl::queue queue_;
 };
 
+template <typename T>
+using unique_usm_ptr = std::unique_ptr<T, usm_deleter<T>>;
+
+inline unique_usm_ptr<void> make_unique_usm(sycl::queue& q,
```

Comment:
Why `_usm` suffix is needed here?

Reply:
I think no

Reply:
I realized that if I remove `_usm`, other methods look strange:
```
make_unique_usm -> make_unique
make_unique_usm_device -> make_unique_device // Misleading, sound like I'm creating unique device
make_unique_usm_shared -> make_unique_shared // std::make_unique or std::make_shared?
make_unique_usm_host -> make_unique_host // Also not clear
```